crates/uv/tests/it/pip_compile.rs

```diff
@@ -15402,3 +15402,31 @@ fn respect_non_local_preference() -> Result<()> {
 
     Ok(())
 }
+
+#[cfg(unix)]
+#[test]
+fn compile_broken_active_venv() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str("anyio==3.7.0")?;
+
+    // Simulate a removed Python interpreter
+    fs_err::remove_file(context.interpreter())?;
+    std::os::unix::fs::symlink("/removed/python/interpreter", context.interpreter())?;
+
+    uv_snapshot!(context
+        .pip_compile()
+        .arg("requirements.in"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: Failed to inspect Python interpreter from active virtual environment at `.venv/bin/python3`
+      Caused by: Broken symlink at `.venv/bin/python3`, was the underlying Python interpreter removed?
+
+    hint: To recreate the virtual environment, run `uv venv`
```

Comment:
What if the virtual environment was found in a parent directory or specified some other way? I'm worried this hint might be misleading.

Reply:
Should we make the hint more generic, like "hint: The Python interpreter seems to have been uninstalled, consider recreating the virtual environment"?

Reply:
That seems more correct. You could sniff for `pyenv.cfg` too? If this was handled in the caller instead (as it sort of was before this PR?) you would have information about whether it's a virtual environment or not?

Reply:
@jtfmumm you got an opinion on that from the other work on healing broken venvs wrt `pyvenv.cfg`?

Reply:
i.e., we have https://github.com/astral-sh/uv/blob/7d12b6af00fea7b45d0c0e1e2d0f9431b3cf9b9c/crates/uv-python/src/discovery.rs#L819-L828